solution2.py

```diff
@@ -0,0 +1,45 @@
+# // Time Complexity : o(1)
+# // Space Complexity : o(n)
+# // Did this code successfully run on Leetcode : yes
+# // Any problem you faced while coding this : no
+#
+#
+# // Your code here along with comments explaining your approach
+# Problem 2:
+# Design MinStack (https://leetcode.com/problems/min-stack/)
+
+import sys
+
+
+class MinStack:
+
+    def __init__(self):
+        """
+        initialize your data structure here.
+        """
+        self.s1 = []
+        self.min = sys.maxsize
+
+    def push(self, x: int) -> None:
+        if x <= self.min:
+            self.s1.append(self.min)
+            self.min = x
+        self.s1.append(x)
+
+    def pop(self) -> None:
+        val = self.s1.pop()
+        if val == self.min:
+            self.min = self.s1.pop()
+
+    def top(self) -> int:
```

Comment:
pop and top will crash if called on an empty stack, please add error guarding if statements to prevent from operating on an empty stack.